test/storage_test.rb

```diff
@@ -44,4 +44,24 @@ def test_cannot_write_to_history_storage
     history_storage = TTNT::Storage.new(@repo, sha)
     assert_raises { history_storage.write!(@section, @data) }
   end
+
+  def test_storage_file_resides_with_rakefile
+    @storage.write!(@section, @data)
+    subdir = "#{@repo.workdir}/tmp"
+    Dir.mkdir(subdir)
+    rakefiles = ["#{@repo.workdir}/Rakefile", "#{subdir}/Rakefile"]
+    FileUtils.copy rakefiles[0], rakefiles[1]
+
+    TTNT.instance_variable_set(:@root_dir, nil)
```

Comment:
I did not provide setter method since application logic does not write `root_dir` for now. But I realized providing setter does no harm and makes tests simpler so I fixed it: https://github.com/Genki-S/ttnt/commit/a010125e31e2448764865f19c994c8e44a9be9d6
Thank you for pointing out! :smile: